packages/features/bookings/lib/handleCancelBooking.ts

```diff
@@ -452,24 +453,31 @@ async function handler(req: CustomRequest) {
             bookingToDelete.recurringEventId &&
             allRemainingBookings
           ) {
-            const promises = bookingToDelete.user.credentials
-              .filter((credential) => credential.type.endsWith("_calendar"))
-              .map(async (credential) => {
-                const calendar = await getCalendar(credential);
-                for (const updBooking of updatedBookings) {
-                  const bookingRef = updBooking.references.find((ref) => ref.type.includes("_calendar"));
-                  if (bookingRef) {
-                    const { uid, externalCalendarId } = bookingRef;
-                    const deletedEvent = await calendar?.deleteEvent(uid, evt, externalCalendarId);
-                    apiDeletes.push(deletedEvent);
+            if (bookingToDelete.references.length > 0 && bookingToDelete?.references[0].recurringMeetingId) {
+              const recurringMeetingId = bookingToDelete?.references[0].recurringMeetingId;
+              apiDeletes.push(
+                calendar?.deleteEvent(recurringMeetingId, evt, externalCalendarId) as Promise<unknown>
```

Comment:
What if we just want to delete the first occurence and not all the occurences?

Reply:
Then the else condition at line 483 would be triggered. For recurring events, there are two IDs, the recurring event ID and the instance ID. I've set the instance ID for the existing `UID`, `meetingID` fields in the `BookingReference` database and created a new field for the `recurring ID`. Therefore the existing flows for modifying bookings will affect only that particular instance and not recurring event as a whole.